sensor/src/hardware/co2_sensor.py

```diff
@@ -31,29 +32,32 @@ def __init__(
         self,
         config: custom_types.Config,
         testing: bool = False,
+        simulate: bool = False,
     ) -> None:
         self.logger = utils.Logger(
             origin="co2-sensor",
             print_to_console=testing,
             write_to_file=(not testing),
         )
         self.config = config
+        self.simulate = simulate
         self.logger.info("Starting initialization.")
         self.last_powerup_time: float = time.time()
 
-        # power pin to power up/down CO2 sensor
-        self.pin_factory = utils.get_gpio_pin_factory()
-        self.power_pin = gpiozero.OutputDevice(
-            pin=CO2_SENSOR_POWER_PIN_OUT, pin_factory=self.pin_factory
-        )
+        if not simulate:
```

Comment:
here you could probably:
if simulate:
    self.logger.info("Finished simulated initialization.")
    return

Reply:
agreed, easier to read this way